src/pcre2_jit_compile.c

```diff
@@ -5687,11 +5714,44 @@ PCRE2_UCHAR othercase[1];
 #endif
 
 repeat = 1;
+stack_ptr = 0;
+step_count = 10000;
 while (TRUE)
   {
-  if (*rec_count == 0)
+  if (--step_count == 0)
     return 0;
-  (*rec_count)--;
+
+  SLJIT_ASSERT(chars <= chars_start + MAX_N_CHARS);
+
+  if (chars >= chars_end)
+    {
+    if (stack_ptr == 0)
+      return chars_end - chars_start;
+
+    --stack_ptr;
+    cc = cc_stack[stack_ptr];
+    chars = chars_stack[stack_ptr];
+
+    if (chars >= chars_end)
+      continue;
+
+    if (next_alternative_stack[stack_ptr] != 0)
+      {
+      /* When an alternative is processed, the
+      next alternative is pushed onto the stack. */
+      SLJIT_ASSERT(*cc == OP_ALT);
+      alternative = cc + GET(cc, 1);
+      if (*alternative == OP_ALT)
+        {
+        SLJIT_ASSERT(stack_ptr < SCAN_PREFIX_STACK_END);
+        cc_stack[stack_ptr] = alternative;
+        chars_stack[stack_ptr] = chars;
+        next_alternative_stack[stack_ptr] = 1;
```

Comment:
The assignments to `chars_stack` and `next_alternative_stack` should be setting the same values that we just read.

That's just an observation. I don't have a preference between keeping the redundant assignments as they are here, or converting them to a debug assertion.

Reply:
I haven't noticed that. This is a quite good observation.